test/build.py

```diff
@@ -11,9 +11,9 @@
         os.unlink(cpp_test_bin)
 else:
     test_env['LIBS'] = [env['MAPNIK_NAME']]
-    test_env.AppendUnique(LIBS=copy(env['LIBMAPNIK_LIBS']))
     test_env.AppendUnique(LIBS='mapnik-wkt')
     test_env.AppendUnique(LIBS='mapnik-json')
+    test_env.AppendUnique(LIBS=copy(env['LIBMAPNIK_LIBS']))
```

Comment:
Yup, `libmapnik-json` refers to `icu_52::UnicodeString::~UnicodeString()`, which is defined in `libicuuc`. With the previous link order, `-licuuc` preceded `-lmapnik-json` and so I was getting:

    /usr/bin/ld: src/libmapnik.so: undefined reference to symbol '_ZN6icu_5213UnicodeStringD1Ev'
    /usr/lib/x86_64-linux-gnu/libicuuc.so: error adding symbols: DSO missing from command line
    collect2: error: ld returned 1 exit status

Swapping the link order means that the ICU symbol is available.